csharp/src/Drivers/Apache/Spark/SparkConnection.cs

```diff
@@ -269,32 +271,58 @@ protected override async ValueTask<TProtocol> CreateProtocolAsync()
+            properties.TryGetValue(SparkParameters.Port, out string? port);
+
+            Uri baseAddress = GetBaseAddress(hostName, scheme, path, port);
+            AuthenticationHeaderValue authenticationHeaderValue = GetAuthenticationHeaderValue(authType, token, username, password);
+
+            HttpClient httpClient = new();
+            httpClient.BaseAddress = baseAddress;
+            httpClient.DefaultRequestHeaders.Authorization = authenticationHeaderValue;
             httpClient.DefaultRequestHeaders.UserAgent.ParseAdd(UserAgent);
             httpClient.DefaultRequestHeaders.AcceptEncoding.Clear();
             httpClient.DefaultRequestHeaders.AcceptEncoding.Add(new StringWithQualityHeaderValue("identity"));
             httpClient.DefaultRequestHeaders.ExpectContinue = false;
 
-            TConfiguration config = new TConfiguration();
-
-            ThriftHttpTransport transport = new ThriftHttpTransport(httpClient, config);
+            TConfiguration config = new();
+            ThriftHttpTransport transport = new(httpClient, config);
             // can switch to the one below if want to use the experimental one with IPeekableTransport
             // ThriftHttpTransport transport = new ThriftHttpTransport(httpClient, config);
             await transport.OpenAsync(CancellationToken.None);
             return new TBinaryProtocol(transport);
         }
 
+        private static AuthenticationHeaderValue GetAuthenticationHeaderValue(string? authType, string? token, string? username, string? password) =>
+            !string.IsNullOrEmpty(token)
+            && (string.IsNullOrEmpty(authType) || SparkAuthTypeConstants.AuthTypeToken.Equals(authType, StringComparison.OrdinalIgnoreCase))
+                ? new AuthenticationHeaderValue("Bearer", token)
+                : !string.IsNullOrEmpty(username) && !string.IsNullOrEmpty(password)
+                  && (string.IsNullOrEmpty(authType) || SparkAuthTypeConstants.AuthTypeBasic.Equals(authType, StringComparison.OrdinalIgnoreCase))
+                    ? new AuthenticationHeaderValue("Basic", Convert.ToBase64String(Encoding.UTF8.GetBytes($"{username}:{password}")))
+                    : throw new AdbcException("Missing connection properties. Must contain 'token' or 'username' and 'password'");
+
+        private static Uri GetBaseAddress(string? hostName, string? scheme, string? path, string? port)
+        {
+            bool isPortSet = !string.IsNullOrEmpty(port);
+            bool isValidPortNumber = int.TryParse(port, out int portNumber) && portNumber > 0;
+            bool isDefaultHttpsPort = !isPortSet || (isValidPortNumber && portNumber == 443);
+            string uriScheme = Uri.CheckSchemeName(scheme)
+                ? scheme!
+                : isDefaultHttpsPort
+                    ? Uri.UriSchemeHttps
+                    : Uri.UriSchemeHttp;
+            int uriPort = isValidPortNumber ? portNumber : -1;
```

Comment:
shouldn't an invalid port produce an error instead of picking a default value?

Reply:
Good catch. Changed to throw an error if port number is invalid/out of range.